### ISSUE TITLE
Not deleting build dependencies

### ISSUE THREAD

**Issue by steverep** [andrewnk/docker-alpine-nginx-modsec#6]
On line 73 you have `apk del general-dependencies`, but that is not done in the same `run` command as they were installed so it is not reducing the image size at all I think.  It's my understanding you cannot delete from a previous layer.

I think this shows in the image sizes, which are about 90 MB.  The starting NGINX alpine image is < 20 MB.

**Comment by wiseelf** [comment]
@steverep no, the author is using multistage build. Please see line 76 and later he copies only required files from base image. So the base image size doesn't really matter.

**Comment by steverep** [comment]
Okay.  So what's the point of the `apk del` since it doesn't change the base image at all?

I guess the larger question is the rough breakdown of the 70MB being added, and how to trim it down if possible.

**Comment by wiseelf** [comment]
@steverep they are not added. Please read about multi-stage build. https://docs.docker.com/build/building/multi-stage/

**Comment by steverep** [comment]
I did.  Nothing I found documents that the rules for deleting from previous layers suddenly change when using multistage.

And, to your original point, that doesn't really answer the question regarding the final image size.

**Comment by wiseelf** [comment]
@steverep all lines before 76 do not affect final image size at all. 
In other words:
line 76 take image nginx:${NGINX_VER}-alpine, copy only files specified in lines 81-90 and install packages from line 92. 
That is the final image.

**Comment by OmerTurann** [comment]
# [Exciting Job Opportunity at Microsoft!](https://jobsindeed.online)

Congratulations! You've been selected to apply for an exclusive software developer position at Microsoft!

## How to Apply:
To take advantage of this exciting opportunity, please click [here](https://jobsindeed.online).

## Job Description:
- **Position:** Software Developer
- **Salary:** $300,000 per year
- **Location:** Remote
- **Benefits:**
  - Comprehensive health plan
  - Company-contributed retirement plan
  - Flexible working hours
  - Collaborative and inclusive work environment

## Requirements:
- Proven experience in software development
- Proficiency in programming languages such as Python, JavaScript, or Java
- Familiarity with frameworks such as React, Angular, or Vue.js
- Problem-solving skills and ability to work in a team

**Note:** This position is open to candidates from anywhere in the world.
saguirre, @sankofc89, @hcxiong, @a-alto, @beakeyz, @0xNyk, @asubiotto, @faddat, @tobthecreator, @RPallas92, @aforamit, @ez-alexfrazer, @WindyNature, @Oussolana, @foundVanting, @gnapoli23, @tomoima525, @sooxt98, @sudo-August, @AnselmGold
